build-system/server.js

```diff
@@ -79,7 +79,28 @@ app.use('/api/echo/post', function(req, res) {
   res.end(JSON.stringify(req.body, null, 2));
 });
 
+const ORIGIN_REGEX = new RegExp('^http://localhost:8000|' +
+    '^https?://.+\.herokuapp\.com:8000');
+const SOURCE_ORIGIN_REGEX = new RegExp('^http://localhost:8000|' +
```

Comment:
Please add a comment that in practice, this would be the publisher's origin, and maybe a link to the CORS spec.